src/components/navigation/navigationItem.svelte

```diff
@@ -0,0 +1,152 @@
+<script lang="ts">
+  import { createEventDispatcher, onMount } from 'svelte'
+  import type { SvelteHTMLElements } from 'svelte/elements'
+  import type { IconName } from '../../../icons/meta'
+  import Icon from '../icon/icon.svelte'
+
+  // This black magic comes from this thread:
+  // https://github.com/sveltejs/language-tools/issues/442#issuecomment-1278618531
+  //
+  // To quote that thread - This is "absolute bonkers!"
+  //
+  // It's interesting, minor variations which I would expect to work on don't,
+  // and this is the only combination which seems to do what we want and I'm not
+  // clear on why. You're welcome to try other approaches here.
+  //
+  // Tips, for if things aren't working right:
+  // 1) npm run gen-types
+  // 2) Reload your VSCode Window (sometimes the Svelte Type Checker struggles).
+  // 3) Make sure any script tags on your component have a `lang="ts"` attribute.
+  type Href = $$Generic<string | undefined>
+  type Disabled = $$Generic<undefined extends Href ? boolean : undefined>
+  type ExcludedProps = 'size' | 'href' | 'hreflang'
+
+  type CommonProps = {
+    inList?: boolean
+    icon?: IconName
+  }
+
+  type ButtonProps = CommonProps &
+    Omit<Partial<SvelteHTMLElements['button']>, ExcludedProps> & {
+      isDisabled?: Disabled
+      isLoading?: boolean
+      href?: never
+    }
+
+  type LinkProps = CommonProps &
+    Omit<Partial<SvelteHTMLElements['a']>, ExcludedProps> & {
+      href: Href
+      isCurrent?: boolean
+    }
+
+  type $$Props = LinkProps | ButtonProps
+
+  export let href: Href = undefined
+  export let icon: IconName = undefined
+  export let isLoading: boolean = false
+  export let isDisabled: boolean = false
+  export let isCurrent: boolean = window.location.pathname === href
+  export let inList: boolean = true
+
+  const checkIfCurrent = () => {
+    isCurrent = window.location.pathname === href
+  }
+
+  $: tag = href ? 'a' : ('button' as 'a' | 'button')
+
+  /**
+   * Optional click handler
+   */
+  const dispatch = createEventDispatcher()
+  function onClick(event) {
+    dispatch('click', event)
+  }
+
+  onMount(() => {
+    ;['pushState', 'replaceState'].forEach((name) => {
+      const original = history[name]
+      history[name] = function () {
+        original.apply(history, arguments)
+        checkIfCurrent()
+      }
+    })
+
+    window.addEventListener('popstate', checkIfCurrent)
```

Comment:
Might want to listen to `hashchange` too? Do we need to clean these up when the component gets unmounted too?